singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt

```diff
@@ -49,26 +49,24 @@ class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
         loginParams: LoginParams,
         context: Context? = null,
         sessionTime: Long = 86400
-    ): CompletableFuture<TorusKey> {
-        val torusKeyCompletableFuture: CompletableFuture<TorusKey> =
-            CompletableFuture<TorusKey>()
+    ): CompletableFuture<SFAKey> {
+        val torusKeyCompletableFuture: CompletableFuture<SFAKey> =
+            CompletableFuture<SFAKey>()
 
-        val details: NodeDetails =
+        val nodeDetails: NodeDetails =
             nodeDetailManager.getNodeDetails(loginParams.verifier, loginParams.verifierId)
                 .get()
         val pubDetails: TorusPublicKey = torusUtils.getUserTypeAndAddress(
-            details.torusNodeEndpoints,
-            details.torusNodePub,
-            VerifierArgs(loginParams.verifier, loginParams.verifierId),
-            true
-        ).get()
-        if (pubDetails.upgraded) {
-            val response: CompletableFuture<TorusKey> = CompletableFuture<TorusKey>()
+            getTorusNodeEndpoints(nodeDetails), loginParams.verifier,
+            loginParams.verifierId, null
+        )
+        if (pubDetails.metadata.isUpgraded) {
+            val response: CompletableFuture<SFAKey> = CompletableFuture<SFAKey>()
             response.completeExceptionally(Exception(SFAError.getError(ErrorCode.USER_ALREADY_ENABLED_MFA)))
             return response
         }
-        val retrieveSharesResponse = getRetrieveSharesResponse(loginParams, details, pubDetails)
-        if (retrieveSharesResponse.privKey == null) {
+        val retrieveSharesResponse = getRetrieveSharesResponse(loginParams, nodeDetails, pubDetails)
+        if (retrieveSharesResponse.finalKeyData.privKey == null) {
```

Comment:
This is not going to work
```
        if (retrieveSharesResponse.finalKeyData.privKey == null) {
```

See here:
https://github.com/torusresearch/torus-utils-java/pull/28/files#diff-45201af8a5356f716eac4e63a4b783a84262d3d66bb2c9de18e9b95d4930524dR12

And here:
https://github.com/torusresearch/torus-utils-java/pull/28/files#diff-b989ac57484b4c05fd81ef8d37f334f9a318f7a9ba99e4f3e825e891c3e6cbc9R13


```suggestion
        if (retrieveSharesResponse.getFinalKeyData().getPrivKey() == null) {
```

Apply to all similar.